crates/ty_python_semantic/src/types/call/arguments.rs

```diff
@@ -207,16 +182,10 @@ impl<'a, 'db> CallArgumentTypes<'a, 'db> {
     }
 }
 
-impl<'a> Deref for CallArgumentTypes<'a, '_> {
-    type Target = CallArguments<'a>;
-    fn deref(&self) -> &CallArguments<'a> {
-        &self.arguments
-    }
-}
-
-impl<'a> DerefMut for CallArgumentTypes<'a, '_> {
-    fn deref_mut(&mut self) -> &mut CallArguments<'a> {
-        &mut self.arguments
+impl<'a> From<Vec<Argument<'a>>> for CallArguments<'a, '_> {
+    fn from(arguments: Vec<Argument<'a>>) -> Self {
+        let types = vec![None; arguments.len()];
+        Self { arguments, types }
     }
 }
```

Comment:
NIt: I think I would prefer an explicit method on `CallArguments` instead of using `From` here which makes it clear that all types will be uninitialized.